shoop/configuration.py

```diff
@@ -0,0 +1,125 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shoop.
+#
+# Copyright (c) 2012-2015, Shoop Ltd. All rights reserved.
+#
+# This source code is licensed under the AGPLv3 license found in the
+# LICENSE file in the root directory of this source tree.
+"""
+API for shoop dynamic configurations. Values is stored to database in
+`shoop.core.configurations.ConfigurationItem model`. Shop specific
+configurations is cached.
+
+Shop can be None which means that configuration item is global. For shop
+first all global configurations is read and then shop specific
+configuration which may override global configurations.
+"""
+from __future__ import unicode_literals
+
+from shoop.core import cache
+from shoop.core.models import ConfigurationItem
+
+
+def set(shop, key, data):
+    """
+    Set configuration item value for a shop or globally.
+
+    If given `shop` is ``None``, the value of given `key` is set
+    globally for all shops.  Otherwise sets a shop specific value which
+    overrides the global value in configuration of the specified shop.
+
+    :param shop: Shop to set value for
+    :type shop: shoop.core.models.Shop|None
+    :param key: Name of the key to set
+    :type key: str
+    :param value: Value to set.  Note: Must be JSON serializable.
+    :type value: Any
+    """
+    conf, _ = ConfigurationItem.objects.update_or_create(shop=shop, key=key, defaults={"value": data})
```

Comment:
Sorry, I was wrong about this. Didn't notice that the call was `update_or_create` (thought it was `get_or_create`). Anyway.. the conf is unneeded and so the return value may be totally discarded.